internal/monitors/statsd/monitor.go

```diff
@@ -0,0 +1,137 @@
+package statsd
+
+import (
+	"context"
+	"time"
+
+	"github.com/signalfx/golib/datapoint"
+	"github.com/signalfx/golib/sfxclient"
+	"github.com/signalfx/signalfx-agent/internal/core/config"
+	"github.com/signalfx/signalfx-agent/internal/monitors"
+	"github.com/signalfx/signalfx-agent/internal/monitors/types"
+	"github.com/signalfx/signalfx-agent/internal/utils"
+	log "github.com/sirupsen/logrus"
+)
+
+const monitorType = "statsd"
+
+var logger = utils.NewThrottledLogger(log.WithFields(log.Fields{"monitorType": monitorType}), 30*time.Second)
+var golibLogger = &utils.LogrusGolibShim{FieldLogger: logger.FieldLogger}
+
+func init() {
+	monitors.Register(monitorType, func() interface{} { return &Monitor{} }, &Config{})
+}
+
+type converterInput struct {
+	Pattern    string
+	Metric     string
+	Dimensions []string
+}
+
+// Config for this monitor
+type Config struct {
+	config.MonitorConfig `yaml:",inline" acceptsEndpoints:"false" singleInstance:"true"`
+	// The host/address on which to bind the UDP listener that accepts statsd
+	// datagrams
+	ListenAddress string `yaml:"listenAddress" default:"localhost"`
+	// The port on which to listen for statsd messages
+	ListenPort uint16 `yaml:"listenPort" default:"8125"`
+	// The protocol for statsd communications
+	Protocol string `yaml:"protocol" default:"udp"`
```

Comment:
Instead of a free-form string config field, it is easier for users to have a bool field called `listenTCP` that defaults to `false`, but if set to true would listen on a TCP socket.